src/main/java/profile_screens/ProfileScreen.java

```diff
@@ -1,75 +1,147 @@
                 panel.add(restarauntName);
                 panel.add(stars);
                 panel.add(reviewText);
 
-                review_panels.add(panel);
+            reviewsPanel.add(panel);
 
-                totalStars = totalStars + starCount;
-            }
+            totalStars = totalStars + starCount;
+        }
+
+
+        JLabel restaurantsLabel = new JLabel("Restaurants");
+        JPanel restaurantsPanel = new JPanel();
 
             int averageStars = totalStars / this.reviews.size();
 
 
-            JLabel averageStarLabel = new JLabel(Integer.toString(averageStars));
+        for (Restaurant restaurant : this.restaurants) {
+            JPanel tempPanel = new JPanel();
+            JLabel restaurantName = new JLabel(restaurant.getName());
+            JLabel restaurantLocation = new JLabel(restaurant.getLocation());
+            JButton viewButton = new JButton("View");
+            ViewRestaurantActionListener viewRestaurantActionListener = new ViewRestaurantActionListener();
+            viewButton.addActionListener(viewRestaurantActionListener);
+            if (isLoggedIn) {
+                JButton editButton = new JButton("Edit");
+                EditRestaurantActionListener editRestaurantActionListener = new EditRestaurantActionListener();
+                editButton.addActionListener(editRestaurantActionListener);
+            }
 
-            profile_screen_window.add(title);
-            profile_screen_window.add(averageStarLabel);
+            tempPanel.add(restaurantName);
+            tempPanel.add(restaurantLocation);
+            tempPanel.add(viewButton);
+            restaurantsPanel.add(tempPanel);
+        }
 
+        if (isLoggedIn && isOwner) {
+            JButton createRestaurantButton = new JButton("Create Restaurant");
+            CreateRestaurantActionListener createRestaurantActionListener = new CreateRestaurantActionListener();
+            createRestaurantButton.addActionListener(createRestaurantActionListener);
+        }
 
-            for (JPanel panel : review_panels) {
-                profile_screen_window.add(panel);
-            }
+        int averageStars = totalStars / this.reviews.size();
+
+        JLabel averageStarLabel = new JLabel(Integer.toString(averageStars));
+
+        JLabel averageStarLabel = new JLabel(Integer.toString(averageStars));
+
+        JLabel reviewsLabel = new JLabel("Reviews");
+        profile_screen_window.add(reviewsLabel);
+        profile_screen_window.add(reviewsPanel);
+        profile_screen_window.add(restaurantsLabel);
+        profile_screen_window.add(restaurantsPanel);
+
+
+        for (JPanel panel : review_panels) {
+            profile_screen_window.add(panel);
+        }
 
             profile_screen_window.setVisible(true);
         }catch(IOException e){
             System.out.println("An error occurred. Please try again later.");
         }
     }
 
+    @Override
+    public void actionPerformed(ActionEvent e) {
+
```

Comment:
If you don't intend to use "this" as an action listener, you don't have to have this class implement action listener. Otherwise looks good

Reply:
Good catch, will change.